docs/utils/atom-with-suspense.mdx

```diff
@@ -0,0 +1,12 @@
+---
+title: atomWithSuspense
+---
+
+## Usage
+
+```ts
+atomWithSuspense<Value>(): WritableAtom<Value | Promise<Value>, SetStateAction<Value>
+```
+
+This creates a new atom that suspend by default.
+When the value is set for the first time it will act like a normal atom.
```

Comment:
We cover this case with this pattern:
```js
const anAtom = atom(new Promise<Value>(() => {}))
```

I think this is probably a doc issue. Feel free to open a new PR for improving docs.

Having that said, providing a shorthand with an explicit function name can be a good idea. This can fall into my 3rd-party package: https://github.com/jotai-labs/jotai-suspense Would you be interested in contributing there? I would maybe call it `atomWithPending`.

Reply:
@eolme If you are willing to contribute, my suggestions are

a) in this `jotai` repo, open a PR to improve docs to illustrate this infinite pending usage. A codesandbox example would be nice to be included.
b) in the other `jotai-suspense` repo, open an issue or PR to add `atomWithPending` which should be as simple as follows, but not tested. Also an example would be nice. (you are generally welcome to contribute in the `jotai-suspense` repo, as it's almost empty. 😄 Especially, README.)

```js
// in JS
const atomWithPending = () => atom(new Promise(() => {}))
```